neo4j/templates/_helpers.tpl

```diff
@@ -96,7 +96,12 @@ If no password is set in `Values.neo4j.password` generates a new random password
     {{- end -}}
     {{- $ignored := set .Values.neo4j "password" $password }}
   {{- end -}}
+  {{- if not .Values.neo4j.passwordFromSecret | and .Values.neo4j.password  -}}
   {{- .Values.neo4j.password }}
+{{- end -}}
+  {{- if .Values.neo4j.passwordFromSecret  -}}
+{{- printf "$(kubectl get secret %s -o go-template='{{.data.NEO4J_AUTH | base64decode }}' | cut -d '/' -f2) " .Values.neo4j.passwordFromSecret -}}
+{{- end -}}
```

Comment:
Considering this was provided independently, do we really want to print it, and risk having it exposed in logs for example ?

Reply:
That's a good point. This only prints the command to run, not the actual value eg;

```
Your release "server1" has been installed  in namespace "neo4j".

The neo4j user's password has been set to "$(kubectl get secret test-auth -o go-template='{{.data.NEO4J_AUTH | base64decode }}' | cut -d '/' -f2) ".To view the progress of the rollout try:

  $ kubectl --namespace "neo4j" rollout status --watch --timeout=600s statefulset/server1

Once rollout is complete you can log in to Neo4j at "neo4j://server1.neo4j.svc.cluster.local:7687". Try:

  $ kubectl run --rm -it --namespace "neo4j" --image "neo4j:5.1.0-enterprise" cypher-shell \
     -- cypher-shell -a "neo4j://server1.neo4j.svc.cluster.local:7687" -u neo4j -p "$(kubectl get secret test-auth -o go-template='{{.data.NEO4J_AUTH | base64decode }}' | cut -d '/' -f2) "
```

Reply:
Indeed I read a bit fast the templating and got tricked by the comment in the description. Thanks !